src/types/acls.rs

```diff
@@ -0,0 +1,398 @@
+use crate::error::{Error, ParseError};
+use crate::parse::try_handle_unilateral;
+use crate::types::UnsolicitedResponse;
+use imap_proto::types::AclRight;
+use imap_proto::Response;
+use ouroboros::self_referencing;
+use std::borrow::Cow;
+use std::collections::HashSet;
+use std::fmt::{write, Display, Formatter};
+use std::sync::mpsc;
+
+/// enum used for set_acl to specify how the ACL is to be modified.
+pub enum AclModifyMode {
+    /// Replace all ACLs on the identifier for the mailbox
+    Replace,
+    /// Add the given ACLs to the identifier for the mailbox
+    Add,
+    /// Remove the given ACLs from the identifier for the mailbox
+    Remove,
+}
+
+/// Helpful wrapper around the ACL rights vector
+#[derive(Debug, Eq, PartialEq)]
+pub struct AclRightList {
```

Comment:
Would this be better as just `AclRights`? Feels weird to call it a list when it's a set.